app/src/main/scala/uk/ac/wellcome/platformapi/models/Record.scala

```diff
@@ -0,0 +1,26 @@
+package uk.ac.wellcome.platform.api.models
+
+import com.sksamuel.elastic4s.searches.RichSearchHit
+
+
+case class Record(
+  altRefNo: String,
+  title: String,
+  materialType: String,
+  date: Option[String],
+  acquisition: Option[String]
+)
+case object Record {
+  def apply(hit: RichSearchHit): Record = {
+    val data: Map[String, AnyRef] =
+      hit.sourceAsMap.filter(o => o._2 != null)
+
+    Record(
+      altRefNo=data("AltRefNo").toString,
+      title=data("Title").toString,
+      materialType=data("Material").toString,
+      date=data.get("Date").map(_.toString),
+      acquisition=data.get("Acquisition").map(_.toString)
```

Comment:
What’s the difference between `foo.toString` and `foo.map(_.toString)`? Is it something to do with futures (and if so, why are only some of the fields futures?). Or because some of the fields are absent/null?